src/main/java/org/hiero/sketch/spreadsheet/Bucket.java

```diff
@@ -0,0 +1,43 @@
+package org.hiero.sketch.spreadsheet;
+import java.util.*;
+/**
+ * A Bucket to be used as part of a histogram.
+ */
+public class Bucket<T> {
```

Comment:
I guess using doubles for all histograms is possible, though especially with a discrete domain may be somewhat awkward. I had each bucket have its own boundaries because I though we may want to cover the case buckets are not disjoint, but if that never happens design should be simpler. Multi-dimensional buckets are just 'AND' of one dimensional buckets. I would assume this is how we want to design them. We can discuss this further in a meeting.